### ISSUE TITLE
oncePer option bug

### ISSUE THREAD

**Issue by studypyth** [argoproj-labs/argocd-notifications#183]
Hi,All, i have issue with duplicate notifications, i am getting for each commit notification twice  I am using oncePer option, but i have no affect to problem.  Also this my configuration of trigger
```
  trigger.on-sync-running: |
    - when: app.status.operationState.phase in ['Running']
      oncePer: app.status.sync.revision
      description: Application is being synced
      send: [app-sync-running]

  trigger.on-sync-succeeded: |
    - description: Application syncing has succeeded
      oncePer: app.status.sync.revision
      send:
      - app-sync-succeeded
      when: app.status.operationState.phase in ['Succeeded'] and app.status.health.status == 'Healthy'
```
And for debug i changed template  to
```
      Debug: {{html .app.metadata.name}}  {{html .app.status.operationState.syncResult.revision}} {{html .app.status.sync.revision}}
```
And i received this messages 2 notifications in telegram for on-sync-running, as i see oncePer option wasnt worked

```
Debug: develop-devops-loki  <no value> 35f3d95bdfe235ebfc9e736fa2ebb0945d6edfa9
Debug: develop-devops-loki  <no value> 35f3d95bdfe235ebfc9e736fa2ebb0945d6edfa9
```
And duplicate message for on-sync-succeeded
```
Debug: develop-devops-loki  35f3d95bdfe235ebfc9e736fa2ebb0945d6edfa9 35f3d95bdfe235ebfc9e736fa2ebb0945d6edfa9
Debug: develop-devops-loki  35f3d95bdfe235ebfc9e736fa2ebb0945d6edfa9 35f3d95bdfe235ebfc9e736fa2ebb0945d6edfa9
```

**Comment by studypyth** [comment]
@alexmt pls review this issue, I'm strongly sure that this is a bug

**Comment by alexmt** [comment]
Hello @studypyth ,

I think this bug is fixed by https://github.com/argoproj-labs/argocd-notifications/pull/157/files .

**Comment by studypyth** [comment]
> Hello @studypyth ,
> 
> I think this bug is fixed by https://github.com/argoproj-labs/argocd-notifications/pull/157/files .

Hello, i tried to use this option, it also sent duplicate message.

May be my problem related to this: [fix: update cached informer object instead of reloading app to avoid …](https://github.com/argoproj-labs/argocd-notifications/commit/4dd6b0dcd1b5b080e12a8d989f47bde5002f007e)?

**Comment by TedSpinks** [comment]
Hi, I also have the issue with duplicate notifications. I have tried both oncePer suggestions with the same results:
```
  trigger.sync-operation-change: |
    - when: app.status.operationState.phase in ['Error', 'Failed']
      oncePer: app.status.sync.revision
      send: [codefresh-simple-test-pipeline]
```
and
```
  trigger.sync-operation-change: |
    - when: app.status.operationState.phase in ['Error', 'Failed']
      oncePer: app.status.operationState.syncResult.revision
      send: [codefresh-simple-test-pipeline]
```

Here's an excerpt from the logs of the argocd-notifications-controller pod. You can see that it sends the same event twice, before recognizing that it has already been sent.
```
time="2021-04-01T21:47:09Z" level=info msg="Start processing" app=argocd/express-microservice
time="2021-04-01T21:47:09Z" level=info msg="Trigger sync-operation-change result: [{[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk 8623da4cb1e06a79d01532ff39614c2a775a7bf2 [codefresh-simple-test-pipeline] true}]" app=argocd/express-microservice
time="2021-04-01T21:47:09Z" level=info msg="Sending notification about condition 'sync-operation-change.[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk' to '{codefresh }'" app=argocd/express-microservice
time="2021-04-01T21:47:10Z" level=info msg="Processing completed" app=argocd/express-microservice
time="2021-04-01T21:47:10Z" level=info msg="Start processing" app=argocd/express-microservice
time="2021-04-01T21:47:10Z" level=info msg="Trigger sync-operation-change result: [{[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk 8623da4cb1e06a79d01532ff39614c2a775a7bf2 [codefresh-simple-test-pipeline] true}]" app=argocd/express-microservice
time="2021-04-01T21:47:10Z" level=info msg="Sending notification about condition 'sync-operation-change.[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk' to '{codefresh }'" app=argocd/express-microservice
time="2021-04-01T21:47:11Z" level=info msg="Processing completed" app=argocd/express-microservice
time="2021-04-01T21:47:11Z" level=info msg="Start processing" app=argocd/express-microservice
time="2021-04-01T21:47:11Z" level=info msg="Trigger sync-operation-change result: [{[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk 8623da4cb1e06a79d01532ff39614c2a775a7bf2 [codefresh-simple-test-pipeline] true}]" app=argocd/express-microservice
time="2021-04-01T21:47:11Z" level=info msg="Notification about condition 'sync-operation-change.[0].H9WjsqG1dKYm6njOZ7yUQYOA1Wk' already sent to '{codefresh }'" app=argocd/express-microservice
```

**Comment by TedSpinks** [comment]
Ah, I think I see what is going on in my case.  I had a CD pipeline doing an `argocd app sync` command from a container image. The image had client version v1.5.5+0fdef48.  When I upgrade to client version v1.8.5+d0f8edf, the double event issue seems to go away.

**Comment by studypyth** [comment]
Solved in [#204]()

**Comment by TedSpinks** [comment]
Excellent! I tested with [v1.10.0](https://github.com/argoproj-labs/argocd-notifications/releases/tag/v1.1.0) (includes PR [#204](https://github.com/argoproj-labs/argocd-notifications/pull/204)) and it did indeed resolve my issue.
